applications/tari_app_grpc/proto/base_node.proto

```diff
@@ -264,3 +271,19 @@ message GetMempoolTransactionsRequest {
 message GetMempoolTransactionsResponse {
     Transaction transaction = 1;
 }
+
+message TransactionStateRequest {
+    Transaction transaction  = 1;
+}
+
+message TransactionStateResponse {
+    TransactionLocation result =1;
+}
+
+enum TransactionLocation {
+    UNKNOWN = 0;
+    UNCONFIRMED = 1;
```

Comment:
Unconfirmed should return 'Mempool' and REORG should probably map to 'NotStored' or 'Mined'. The caller does not care about this internal details. @philipr-za what do you think?

Reply:
This is the route taken in the Base Node RPC service exposed to the wallet because we want to keep the idea of the Reorg pool as an internal idea. So I would think that this should also be the case in the gRPC which is an external interface. I understand that in the test you want to check how the system reacts to Double Spends vs Orphans but that can be done in an internal Rust test while the Cucumber test should be an Integration test that tests the system as if you are interacting with it from the outside as it will be in the normal operation. 

In the Cucumber test you can create a double spend and see that it is rejected correctly.